pkg/manifests/manifests.go

```diff
@@ -196,8 +174,8 @@ func (f *Factory) DNSService(dns *dnsv1alpha1.ClusterDNS) (*corev1.Service, erro
 	}
 	s.Spec.Selector["dns"] = s.Name
 
-	if dns.Spec.ClusterIP != nil {
-		s.Spec.ClusterIP = *dns.Spec.ClusterIP
+	if len(clusterIP) > 0 {
+		s.Spec.ClusterIP = clusterIP
```

Comment:
I must be missing something, didn't `Spec` get deleted? surprised this compiles

Reply:
That's setting `.spec.clusterIP` on the service, not the clusterdns.